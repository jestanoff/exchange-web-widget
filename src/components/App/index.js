import React from 'react';
import Header from '../Header';
import Exchange from '../Exchange';

const App = () => (
  <>
    <Header />
    <Exchange />
  </>
);

export default App;
