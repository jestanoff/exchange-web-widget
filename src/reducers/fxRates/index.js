const initialState = {
  rates: undefined,
};

export default (state = initialState, action = {}) => {
  switch (action.type) {
    case 'ANY':
      return {
        rates: 'any',
      };

    default:
      return state;
  }
};
