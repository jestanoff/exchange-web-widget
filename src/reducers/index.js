import { combineReducers } from 'redux';
import fxRates from 'reducers/fxRates';

export default combineReducers({ fxRates });
