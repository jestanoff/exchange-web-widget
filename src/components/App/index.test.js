import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { reducer } from 'reducers';
import App from './';

function renderWithRedux(ui, { store = createStore(reducer) } = {}) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    // adding `store` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    store,
  };
}

describe('App component', () => {
  test('can render with redux with defaults', () => {
    const { getByTestId, container } = renderWithRedux(<App />);
    console.log(container);
    expect(getByTestId('count-value')).toHaveTextContent('1');
  });

  // test('should render proper markup', () => {
  //   const { container } = render(<App />);

  //   expect(container).toMatchSnapshot();
  // });
});
