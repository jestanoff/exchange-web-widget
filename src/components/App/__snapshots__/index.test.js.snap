// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render proper markup 1`] = `
<div>
  <header
    class="header"
  >
    <h1
      class="heading1"
    >
      Exchange
    </h1>
  </header>
  <main
    class="container"
  >
    <form
      action="/"
      name="exchange"
    >
      <section
        class="container from"
      >
        <div>
          <h2
            class="currency"
          >
            EUR
          </h2>
          <span
            class="balance"
          >
            Balance: 
            €0
          </span>
        </div>
        <input
          class="input from"
          data-test="input-from"
          maxlength="17"
          name="from"
          placeholder="0"
          required=""
          type="tel"
        />
      </section>
      <section
        class="exchangeRateContainer"
      >
        <a
          class="button"
          href="#"
        >
          Swap
        </a>
        <section
          class="container"
        >
          €1 = CHF 1.2039
        </section>
      </section>
      <section
        class="container to"
      >
        <div>
          <h2
            class="currency"
          >
            CHF
          </h2>
          <span
            class="balance"
          >
            Balance: 
            CHF 0
          </span>
        </div>
        <input
          class="input"
          data-test="input-to"
          maxlength="17"
          name="to"
          placeholder="0"
          required=""
          type="tel"
        />
      </section>
      <button
        class="button"
        type="submit"
      >
        Exchange
      </button>
    </form>
  </main>
</div>
`;
